admins/README.md

```diff
@@ -0,0 +1,129 @@
+# Administrators
+
+## Table of Contents
+- [Components](#components)
+- [Architecture](#architecture)
+
+## Components
+The BonnyCI system is made up of the following components:
+  - [Zuul](#zuul)
+    - Server
+    - Launcher
+    - Merger
+    - Web App
+  - [Nodepool](#nodepool)
+    - Server
+    - Builder
+    - Launcher
+    - Deleter
+  - [Logs host](#logs host)
+
+### Zuul
+Zuul is the central component. It determines what jobs to run and when to run them, it constructs the specific git object(s) to test, and it reports the results back to the client. For more information, refer to [zuul's documentation](http://docs.openstack.org/infra/zuul/).
+
+#### Zuul Server
+The zuul server process runs on the zuul host and is responsible for the list of jobs, list of projects, and job pipelines. The pipelines manage job ordering and scheduling. They are configured with source connections for incoming events, triggers to determine which events start pipeline processing, requirements to determine which changes are eligible to enter the pipeline, and reporters to describe the actions to take once a job completes.
+
+The zuul server communicates with the other zuul components using gearman. It communicates with nodepool using gearman, zookeeper, and zeromq.
+
+#### Zuul Merger
+The zuul merger is a gearman client that constructs the specific git refs to run tests against.
+
+We currently run a single zuul merger on the zuul host, but this component is horizontally-scalable and new instances should be run on separate small hosts. Once a zuul merger connects to the zuul server via gearman, it will immediately begin handling requests.
+
+Since the zuul server does not control which merger performs a given merge, all test slaves must have access to all mergers to fetch the constructed git refs.
+
+#### Zuul Launcher
+The zuul launcher is a gearman client that uses ansible to run the specified test job on the specified type of test slave. It populates several environment variables in the test environment so that the test can know what git ref to fetch and what url to fetch it from. The zuul-cloner tool can be used to checkout the correct version of multiple projects based on these environment variables.
+
+We currently run a single zuul launcher on the zuul host, but this component is horizontally-scalable and new instances should be run on separate small hosts. Once a zuul launcher connects to the zuul server via gearman, it will immediately begin handling requests.
```

Comment:
Add a comment in here that every zuul launcher must be able to reach every node pool node.